### ISSUE TITLE
Cannot get the value of a token type 'Number' as a string.

### ISSUE THREAD

**Issue by winzig** [betalgo/openai#266]
**Describe the bug**

Today I'm randomly getting these exceptions when calling `ChatCompletion.CreateCompletion()`.

```
Exception thrown: 'System.AggregateException' in System.Private.CoreLib.dll: 'One or more errors occurred.'
 Inner exceptions found, see $exception in variables window for more details.
 Innermost exception 	 System.InvalidOperationException : Cannot get the value of a token type 'Number' as a string.
   at System.Text.Json.Utf8JsonReader.GetString()
   at System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1.ReadJsonAndSetMember(Object obj, ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1.ReadJsonAndSetMember(Object obj, ReadStack& state, Utf8JsonReader& reader)
   at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
```

This happens to coincide with a status alert from OpenAI indicating elevated errors with GPT. 

<img width="894" alt="screenshot 2023-05-18 at 15 36 54" src="https://github.com/betalgo/openai/assets/127062/053010e9-0142-4928-82e6-fa1dc630eb4a">

When the error happens it seems to be after the API has taken quite a while to respond. I suspect a timeout of some sort, probably on the OpenAI side of things since on my side I've set HttpClient to have a 15 minute timeout, and the error happens after it sits there for only 30-45 seconds. (Plus I would expect to see a timeout exception in the stack if it was on my side, and I don't.)

**Desktop (please complete the following information):**
 - OS: macOS 13.3.1 (a)
 - Language: C#
 - Version: 6.8.6

**Comment by kayhantolga** [comment]
Is there any chance that you could capture the response from OpenAI?

**Comment by winzig** [comment]
Just tried (it happened again), but the request doesn't seem to be exposed by the API at all, even in debugging info, so not able to see it as-is. If it keeps happening and no one here knows why, I'll need to download the repo and try to run it locally with some debugging added

**Comment by kayhantolga** [comment]
I've never had such an issue before, so it's hard to guess what could be causing it without the response data. I'm sorry about that :/

You have two options to choose from: using the repo or using a proxy tool like Laser Cat Eyes. However, even if you choose to use the repo, it will be difficult to catch errors because httpclient attempts to parse the response object directly. Therefore, you may need to modify some code. I suggest going with the second option.

**Comment by kayhantolga** [comment]
Hi @winzig, do you have any updates?

**Comment by winzig** [comment]
No, I tried to get Laser Cat Eyes going but it seems aimed at web apps, and the issue, while it does reoccur, hasn't stopped me from doing what I need to do, so I haven't had time to spend on figuring out why it happens yet.

**Comment by kayhantolga** [comment]
I am going to close this issue. I believe it will be resolved with #277. Please do not hesitate to reopen if the issue is not resolved or if you believe this is a separate issue.
